src/libstore/s3.cc

```diff
@@ -7,11 +8,183 @@
 
 #include <ranges>
 
-namespace nix {
+#if NIX_WITH_S3_SUPPORT
+
+#  include "nix/util/logging.hh"
+#  include "nix/util/finally.hh"
+#  include "nix/util/error.hh"
+#  include "nix/util/split.hh"
+#  include "nix/util/url.hh"
+
+#  include <aws/crt/Api.h>
+#  include <aws/crt/auth/Credentials.h>
+#  include <aws/crt/io/Bootstrap.h>
+
+#  include <condition_variable>
+#  include <mutex>
+#  include <string_view>
+#  include <pthread.h>
 
 using namespace std::string_view_literals;
 
-#if NIX_WITH_S3_SUPPORT
+namespace nix {
+
+// AWS CRT initialization
+static std::once_flag crtInitFlag;
+static bool crtInitialized = false;
+
+// Forward declaration for cleanup function
+void cleanupCredentialProviderCache();
+
+static void initAwsCrt()
+{
+    std::call_once(crtInitFlag, []() {
```

Comment:
I prefer the atomic static initializer trick, i.e.
```
static auto crtInitialized = [&]() {
    ... init code ...
    return true;
}();
```
C++ guarantees this is executed only once, so no `std::once_flag` is needed.